src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,45 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden && prey.health > 0) {
+      prey.health -= 50;
+      prey.checkHealth();
```

Comment:
[CODE PERFORMANCE] After decreasing the prey's health, you should filter the `Animal.alive` array directly here instead of calling `checkHealth` on the prey. This would improve performance by avoiding unnecessary function calls and making the code cleaner. Use the `filter` method to remove any animals with health less than or equal to 0.